xarray/tests/__init__.py

```diff
@@ -25,111 +26,47 @@
 except ImportError:
     import mock
 
-try:
-    import scipy
-    has_scipy = True
-except ImportError:
-    has_scipy = False
-
-try:
-    import pydap.client
-    has_pydap = True
-except ImportError:
-    has_pydap = False
-
-try:
-    import netCDF4
-    has_netCDF4 = True
-except ImportError:
-    has_netCDF4 = False
-
-
-try:
-    import h5netcdf
-    has_h5netcdf = True
-except ImportError:
-    has_h5netcdf = False
-
-
-try:
-    import Nio
-    has_pynio = True
-except ImportError:
-    has_pynio = False
-
-
-try:
-    import dask.array
-    import dask
-    dask.set_options(get=dask.get)
-    has_dask = True
-except ImportError:
-    has_dask = False
-
-
-try:
-    import matplotlib
-    has_matplotlib = True
-except ImportError:
-    has_matplotlib = False
-
 
-try:
-    import bottleneck
-    if LooseVersion(bottleneck.__version__) < LooseVersion('1.1'):
-        raise ImportError('Fall back to numpy')
-    has_bottleneck = True
-except ImportError:
-    has_bottleneck = False
-
-try:
-    import rasterio
-    has_rasterio = True
-except ImportError:
-    has_rasterio = False
-
-try:
-    import pathlib
-    has_pathlib = True
-except ImportError:
+def _importorskip(modname, minversion=None):
     try:
-        import pathlib2
-        has_pathlib = True
+        mod = importlib.import_module(modname)
+        has = True
+        if minversion is not None:
+            if LooseVersion(mod.__version__) < LooseVersion(minversion):
+                raise ImportError('Minimum version not satisfied')
     except ImportError:
-        has_pathlib = False
-
-
-# slighly simpler construction that the full functions.
-# Generally `pytest.importorskip('package')` inline is even easier
-requires_matplotlib = pytest.mark.skipif(
-    not has_matplotlib, reason='requires matplotlib')
-requires_scipy = pytest.mark.skipif(
-    not has_scipy, reason='requires scipy')
-requires_pydap = pytest.mark.skipif(
-    not has_pydap, reason='requires pydap')
-requires_netCDF4 = pytest.mark.skipif(
-    not has_netCDF4, reason='requires netCDF4')
-requires_h5netcdf = pytest.mark.skipif(
-    not has_h5netcdf, reason='requires h5netcdf')
-requires_pynio = pytest.mark.skipif(
-    not has_pynio, reason='requires pynio')
-requires_scipy_or_netCDF4 = pytest.mark.skipif(
-    not has_scipy and not has_netCDF4, reason='requires scipy or netCDF4')
-requires_dask = pytest.mark.skipif(
-    not has_dask, reason='requires dask')
-requires_bottleneck = pytest.mark.skipif(
-    not has_bottleneck, reason='requires bottleneck')
-requires_rasterio = pytest.mark.skipif(
-    not has_rasterio, reason='requires rasterio')
-requires_pathlib = pytest.mark.skipif(
-    not has_pathlib, reason='requires pathlib / pathlib2'
-)
-
+        has = False
+    # TODO: use pytest skip
```

Comment:
can you briefly elaborate on why pytest.skip doesn't work? a link to this issue would be heplful.